There is a build error, see https://github.com/vmware/tdnf/actions/runs/4208360003/jobs/7304317763:
```
[ 21%] Building C object solv/CMakeFiles/tdnfsolv.dir/simplequery.c.o
/build/common/lock.c: In function 'tdnflock_new':
/build/common/lock.c:69:11: error: expected ';' before '}' token
   69 |     sync()
      |           ^
      |           ;
   70 |    }
      |    ~       
make[2]: *** [common/CMakeFiles/common.dir/build.make:160: common/CMakeFiles/common.dir/lock.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
```

Please test your changes before pushing.

--- a/client/defines.h
+++ b/client/defines.h
@@ -22,12 +22,6 @@
 
 #include "config.h"
 
-/*
- * creating this under /var/run because /var/run/lock doesn't exist
- * in fedora docker images and as a result ci fails
- */
-#define TDNF_INSTANCE_LOCK_FILE     "/var/run/.tdnf-instance-lockfile"
-
 typedef enum
 {
     /* this should be a bitmask */

--- a/common/defines.h
+++ b/common/defines.h
@@ -33,3 +33,9 @@
     } while(0)
 
 #define TDNF_DEFAULT_MAX_STRING_LEN       16384000
+
+/*
+ * creating this under /var/run because /var/run/lock doesn't exist
+ * in fedora docker images and as a result ci fails
+ */
+#define TDNF_INSTANCE_LOCK_FILE     "/var/run/.tdnf-instance-lockfile"

--- a/common/lock.c
+++ b/common/lock.c
@@ -54,6 +54,21 @@ static tdnflock tdnflock_new(const char *lock_path, const char *descr)
 	else
 	{
 		lock->openmode = TDNFLOCK_WRITE | TDNFLOCK_READ;
+
+		/* Write out PID into lock file */
+		char pid_buffer[128] = {0};
+		int rsnpf;
+		rsnpf = snprintf(pid_buffer, sizeof(pid_buffer), "%ld\n", (long) getpid());
+
+		if (rsnpf > 0)
+		{
+			int wr;
+			wr = write(lock->fd, pid_buffer, strlen(pid_buffer));
+			if (wr == 0)
+			{
+				sync();
+			}
+		}
 	}
 
 	lock->fdrefs = 1;
@@ -234,6 +249,12 @@ tdnflock tdnflockFree(tdnflock lock)
 	{
 		tdnflock_release(lock);
 		tdnflock_free(lock);
+	}
+
+	if (remove(TDNF_INSTANCE_LOCK_FILE))
+	{
+			pr_err("WARNING: Unable to remove lockfile(%s)\n",
+							TDNF_INSTANCE_LOCK_FILE);
 	}
 
 	return NULL;

